### PR TITLE
optimize: using `go mod init` to upgrade examples

### PR DESCRIPTION
**Pull request by 1046102779**
/cc @daixiang0 @yaron2 

image

I hasn't permission to modify workflow configuration.

----------------------------

Extra doubt:

Can we abstract api `github.com/dapr/dapr/dapr/proto` into `github.com/dapr/protocol`? I think it is necessary.

In this way, each language sdk does not have to rely on the dapr core repository, and the pb protocols of all languages are stored in the protocol repository, which exists as a standard.

### CODE REVIEW

**Comment by daixiang0** [comment]
> Can we abstract api github.com/dapr/dapr/dapr/proto into github.com/dapr/protocol? I think it is necessary.

> In this way, each language sdk does not have to rely on the dapr core repository, and the pb protocols of all languages are stored in the protocol repository, which exists as a standard.

Please create an issue in dapr core repo, I think it is good.

File: Makefile [hunk 1]
```diff
@@ -7,7 +7,7 @@ all: help
 
 .PHONY: tidy
 tidy: ## Updates the go modules
-	go mod tidy -compat=1.17
+	go mod tidy
```
Comment (daixiang0) [inline Makefile:10]: Do we need `-compat=1.18`?

Comment (1046102779) [reply Makefile:10]: yes. I found that the dapr respository was golang1.18.

**Review by yaron2 — APPROVED** [review]
LGTM

**Comment by yaron2** [comment]
Thanks @1046102779

**Pull request merged**